app/src/main/java/org/p2p/wallet/home/ui/main/delegates/hidebutton/HideTokensDelegate.kt

```diff
@@ -0,0 +1,26 @@
+package org.p2p.wallet.home.ui.main.delegates.hidebutton
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import com.hannesdorfmann.adapterdelegates4.AdapterDelegate
+import com.hannesdorfmann.adapterdelegates4.dsl.adapterDelegateViewBinding
+import org.p2p.uikit.model.AnyCellItem
+import org.p2p.wallet.databinding.ItemTokenGroupButtonBinding
+
+private typealias BindListener = ((binding: ItemTokenGroupButtonBinding, item: TokenButtonCellModel) -> Unit)
+
+private val inflateViewBinding = { inflater: LayoutInflater, parent: ViewGroup ->
+    ItemTokenGroupButtonBinding.inflate(inflater, parent, false)
+}
+
+fun tokenButtonDelegate(
+    onBindListener: BindListener? = null,
+): AdapterDelegate<List<AnyCellItem>> =
+    adapterDelegateViewBinding<TokenButtonCellModel, AnyCellItem, ItemTokenGroupButtonBinding>(
+        viewBinding = inflateViewBinding
+    ) {
+        bind {
+            binding.imageViewTokenState.setImageResource(item.visibilityIcon)
+            onBindListener?.invoke(binding, item)
```

Comment:
maybe you can pass click listeners here, instead of just one big listener?

Reply:
it was a common approach but yes we can do whatever we want

as for me i like this way because we make clicks on fragment side and see what is happening